server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -147,7 +158,7 @@ public enum MetadataVersion {
      * <strong>Think carefully before you update this value. ONCE A METADATA VERSION IS PRODUCTION,
      * IT CANNOT BE CHANGED.</strong>
      */
-    public static final MetadataVersion LATEST_PRODUCTION = IBP_4_0_IV3;
+    public static final MetadataVersion LATEST_PRODUCTION = IBP_4_0_IV3; // when do we update this to `IBP_4_1_IV2`+
```

Comment:
Side question...

Reply:
Well, this is still on `4_0` though -- when should it get bumped to `4_1` ?

Reply:
It gets bumped to 4_1 when MV 4_1-IV0 is production ready.

Reply:
Should we remove this comment?

Reply:
> Should we remove this comment?

Yes :) 

> It gets bumped to 4_1 when MV 4_1-IV0 is production ready.

@jolshan -- from our discussion, my understanding is, that we would only add a MV if something is stable (and `IBP_4_1_IV1` added for QfK should actually be removed)? Is ELR not production ready yet? And if not, why was a new MV added for it?

Reply:
We add the MV version when we create the feature, but don't mark the MV as ready until the feature should be enabled by default. 

I think we could remove 4_1 from QfK.

For ELR, the first MV (4.0) was due to a dependency on a new MV record value. The second one (4.1) will be marked as stable soon I believe. But Calvin would know better than me. :)